rust-src/crypto_common/src/encryption.rs

```diff
@@ -23,25 +23,27 @@ impl From<String> for Password {
 impl FromStr for Password {
     type Err = <String as FromStr>::Err;
 
-    fn from_str(s: &str) -> Result<Self, Self::Err> { Ok(Password { password: s.into() }) }
+    fn from_str(s: &str) -> ParseResult<Self, Self::Err> { Ok(Password { password: s.into() }) }
 }
 
 // Helpers for JSON serialization in base64 standard format.
-fn as_base64<A: AsRef<[u8]>, S>(key: &A, serializer: S) -> Result<S::Ok, S::Error>
+fn as_base64<A: AsRef<[u8]>, S>(key: &A, serializer: S) -> ParseResult<S::Ok, S::Error>
```

Comment:
Also here, the correct thing to use is `Result` since we supply both arguments, consistent with how Result from std is used.